apps/files/src/mixins.js

```diff
@@ -144,38 +144,110 @@ export default {
 
       return '<span class="' + cssClass.join(' ') + '">' + string + '</span>'
     },
-    $_ocUpload_addToQue (e) {
+    checkIfBrowserSupportsFolderUpload () {
+      var el = document.createElement('input')
+      el.type = 'file'
+      return typeof el.webkitdirectory !== 'undefined' || typeof el.mozdirectory !== 'undefined' || typeof el.directory !== 'undefined'
+    },
+    checkIfElementExists (element) {
+      let name = element.name || element
+      return this.files.find((n) => {
+        if (n['name'] === name) {
+          return n
+        }
+      })
+    },
+    processDirectoryEntryRecursively (directory) {
+      this.$client.files.createFolder(directory.fullPath).then(() => {
+        let directoryReader = directory.createReader()
+        let ctrl = this
+        directoryReader.readEntries(function (entries) {
+          entries.forEach(function (entry) {
+            if (entry.isDirectory) {
+              ctrl.processDirectoryEntryRecursively(entry)
+            } else {
+              entry.file(file => {
+                ctrl.$_ocUpload(file, entry.fullPath)
+              })
+            }
+          })
+        })
+      })
+    },
+    $_ocUpload_addDropToQue (e) {
+      let items = e.dataTransfer.items || e.dataTransfer.files
+
+      // Fallback
+      if (items instanceof FileList) {
+        this.$_ocUpload_addDirectoryToQue(e)
+      } else {
+        for (let item of items) {
+          item = item.webkitGetAsEntry()
+          if (item.isDirectory) {
+            if (!this.checkIfElementExists(item)) {
+              this.processDirectoryEntryRecursively(item)
+            } else {
+              this.showNotification({
+                title: this.$gettextInterpolate('Upload for %{folder} failed - Folder already exists', { folder: item.name }, true),
```

Comment:
first translate the message and then interpolate ... see in the remaining code base ...